src/main/java/org/mqttbee/mqtt/handler/auth/MqttAuthHandler.java

```diff
@@ -96,8 +95,8 @@ private void writeConnect(
             @NotNull final ChannelHandlerContext ctx, @NotNull final MqttConnect connect,
             @NotNull final ChannelPromise promise) {
 
-        final MqttClientData clientData = MqttClientData.from(ctx.channel());
-        final Mqtt5EnhancedAuthProvider enhancedAuthProvider = getEnhancedAuthProvider(clientData);
+        enhancedAuthProvider = connect.getRawEnhancedAuthProvider();
+        assert enhancedAuthProvider != null;
```

Comment:
For consistency reasons please consider using`Preconditions.checkNotNull` instead of assert.
```
enhancedAuthProvider = Preconditions.checkNotNull(connect.getRawEnhancedAuthProvider(), "Enhanced authentication provider must not be null.");
```